tapir/coop/tests/membership_resignation/test_service.py

```diff
@@ -17,5 +34,124 @@ def setUp(self) -> None:
         self.given_feature_flag_value(feature_flag_membership_resignation, True)
         mock_timezone_now(self, self.NOW)
 
-    def test_todo(self):
-        self.fail("TODO, test service functions.")
+    def test_update_shifts_and_shares_and_pay_out_day_SharesAndSettingPayOutDay_transferredAndSet(self):
+        resignations = [
+            MembershipResignation.ResignationType.BUY_BACK,
+            MembershipResignation.ResignationType.GIFT_TO_COOP,
+            MembershipResignation.ResignationType.TRANSFER,
+        ]
+
+        self.login_as_member_office_user()
+        for resignation in resignations:
+            share_owner = ShareOwnerFactory.create(nb_shares=2)
+            resignation = MembershipResignationFactory.create(
+                share_owner=share_owner,
+                resignation_type=resignation,
+            )
+            MembershipResignationService.update_shifts_and_shares_and_pay_out_day(
+                resignation=resignation,
+            )
+            shares_after_update = resignation.share_owner.share_ownerships
+            match resignation:
```

Comment:
This should be 3 separate test functions, so that if the test fails, the person running the test knows right away which case is wrong.